src/universal/utils/transit.js

```diff
@@ -57,4 +57,32 @@ export default transit.withExtraHandlers([
       : require('url').parse(value)
     ),
   },
-]);
+];
+
+const modifiedHandlers = handlers.withExtraHandlers(extraHandlers);
+
+const reader = transit.reader('json', {
+  handlers: modifiedHandlers.read,
+  cache: false,
+  // Copied from transit-immutable-js. Without this, Error fails to read.
+  mapBuilder: {
+    init() {
+      return {};
+    },
+    add(m, k, v) {
+      // eslint-disable-next-line no-param-reassign
+      m[k] = v;
+      return m;
+    },
+    finalize(m) {
+      return m;
+    },
+  },
+});
+const writer = transit.writer('json', { handlers: modifiedHandlers.write, cache: false });
+
+export default {
+  ...instance.withExtraHandlers(extraHandlers), // added for test cases
+  toJSON: (data) => writer.write(data),
```

Comment:
do these exports inflate the client bundle?
that is, can we separate the persistent server writer from the single-use (?) client reader? 🤔

Reply:
I don't want to do that type of optimization in this PR. We are exporting almost exactly what was exported previously, so it may be a few bytes bigger, but I dont see that as an issue.